src/map/script.c

```diff
@@ -17931,18 +17931,22 @@ BUILDIN(delmonsterdrop) {
 BUILDIN(getmonsterinfo)
 {
 	struct mob_db *monster;
-	int mob_id;
+	int mob_id, type;
 
 	mob_id = script_getnum(st,2);
+	type = script_getnum(st,3);
+
 	if (!mob->db_checkid(mob_id)) {
 		ShowError("buildin_getmonsterinfo: Wrong Monster ID: %i\n", mob_id);
-		if ( !script_getnum(st,3) ) //requested a string
+		if (type == 0 || type == 23) //requested a string
```

Comment:
please add enum with type constants in C side.
And probably map this enum values to script engine. and remove MOB_ constants from constants.conf

Reply:
Instead adding constants in C and constants.conf you can add constant values only C side.
And add mapping between C and script constants, see function script_hardcoded_constants.
Here no constant values, only names. And because this impossible to add mistake in constants mapping.

All constants you can see in doc/constants.md

Reply:
for 1. after add new constants in script_hardcoded_constants, from constant.conf duplicates must be removed, or server will show warnings always. and ci will fail.
for 2. document in script_commands.txt what for all constants need see constants.md
All constants ALREADY in constants.md

Reply:
@AnnieRuru what 4144 is going for here is:
Mapping of constants should be unified between the source code and the scripting engine therefore when an enum is introduced we move the constant definition to the source code in-order to map them with the enum values.

Reply:
ok understood now, making another PR to address this issue